ApiAuthTokenGenerator/V1/Helpers/ValidateTokenHelper.cs

```
@@ -0,0 +1,36 @@
+using Microsoft.IdentityModel.Tokens;
+using System;
+using System.Collections.Generic;
+using System.IdentityModel.Tokens.Jwt;
+using System.Linq;
+using System.Security.Claims;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ApiAuthTokenGenerator.V1.Helpers
+{
+    public static class ValidateTokenHelper
+    {
+        public static List<Claim> ValidateToken(string token)
+        {
+            try
+            {
+                var handler = new JwtSecurityTokenHandler();
+                var validations = new TokenValidationParameters
+                {
+                    ValidateIssuerSigningKey = true,
+                    IssuerSigningKey = new SymmetricSecurityKey(Encoding.ASCII.GetBytes
+                    (Environment.GetEnvironmentVariable("jwtSecret"))),
+                    ValidateIssuer = false,
+                    ValidateAudience = false
+                };
+                var claims = handler.ValidateToken(token, validations, out var tokenSecure);
+                return claims.Claims.ToList();
+            }
+            catch (SecurityTokenInvalidSignatureException)
+            {
+                return null; //token invalid, return null
```

Would it not be better to return an exception message?

---

I think the reason I left it like this in this scenario, is because the ultimate goal of the lambda function is to return "Allow"/"Deny". If it was an API, I would have returned the exception and then catch and throw it in the controller. In this case, returning null so I can do the if check in the use case. If I am to throw the exception, I will have to add extra code in usecase/lambda handler classes to know that if this exception is throw, I have to return "Deny". Hope this makes sense

---

Agree....if it's not a valid token (i.e. it's invalid/expired/errors) then it's a deny - no need for exceptions (except perhaps us logging the fact there's an exception?)